notifications/test_pipeline.py

```diff
@@ -30,5 +29,9 @@ def test_complete_pipeline(self):
         for i in history_list:
             self.assertIsInstance(i, PipelineHistory)
             self.assertIsInstance(i.timestamp, datetime)
+        # Check logging
+        self.assertEqual(len(mock_logging.info.call_args_list), 8)  # Successful delivery means 8 calls to logging info
+        self.assertEqual(len(mock_logging.warning.call_args_list), 0)
```

Comment:
I think what you're doing here (checking counts of messages) is sufficient, but we need a test that shows what happens when a message expires on its way through the queue.